bbcdll/bbc_tcp_server.py

```
@@ -426,13 +421,19 @@ def set_battle_type(battle_type):
+        except (ValueError, TypeError):
+            return {'success': False, 'error': f'Invalid battle type value: {battle_type}'}
+        
+        if idx < 0 or idx >= len(CT.BATTLE_TYPE):
+            return {'success': False, 'error': f'Battle type index out of range: {idx} (valid: 0-{len(CT.BATTLE_TYPE)-1})'}
+        
+        try:
             page.battletype.set(CT.BATTLE_TYPE[idx])
-            _log('info', f'[Battle] Battle type set: {battle_type}')
-            return {'success': True, 'battle_type': battle_type}
+            _log('info', f'[Battle] Battle type set: index={idx}, name={CT.BATTLE_TYPE[idx]}')
+            return {'success': True, 'battle_type_index': idx}
```

Ruff (0.15.11)

[warning] Consider moving this statement to an `else` block

(TRY300)